src/nupic/algorithms/Anomaly.hpp

```diff
@@ -112,10 +113,16 @@ namespace nupic
                        const std::vector<UInt>& predicted,
                        Real64 inputValue=0, UInt timestamp=0);
 
+//helper
+        void setLikelihood(AnomalyLikelihood newInstance) {
+          this->likelihood_ = newInstance;
+        }
+
       private:
         AnomalyMode mode_;
         Real32 binaryThreshold_;
         std::unique_ptr<nupic::util::MovingAverage> movingAverage_;
+        AnomalyLikelihood likelihood_; //TODO which params/how pass them to constructor?
```

Comment:
This should be a pointer. Then we should create it in the constructor, but only for AnomalyModes that use it. Then delete it in the destructor.